pandas/tests/series/test_sorting.py

```diff
@@ -144,3 +144,18 @@ def test_sort_index_multiindex(self):
         # rows share same level='A': sort has no effect without remaining lvls
         res = s.sort_index(level='A', sort_remaining=False)
         assert_series_equal(s, res)
+
+    def test_sort_index_nan(self):
+
+        # GH13729
+        ser = Series(['A', np.nan, 'C', 'D'], [1, 2, 0, np.nan])
+
+        # na_position='last', kind='quicksort'
+        result = ser.sort_index(kind='quicksort', na_position='last')
+        expected = Series(['C', 'A', np.nan, 'D'], [0, 1, 2, np.nan])
+        assert_series_equal(result, expected)
+
+        # na_position='first'
+        result = ser.sort_index(na_position='first')
```

Comment:
I just checked, this is the error it returns:

```
TypeError: sort_index() got an unexpected keyword argument 'random_argument'
```

Just to make sure we are in sync, this is what I need to add test for, right? @jorisvandenbossche